test/cli/parse-tree-json-with-locs/parse-tree-json-with-locs.out

```diff
@@ -0,0 +1,132 @@
+{
+  "type" : "Module",
+  "loc" : "test/cli/parse-tree-json-with-locs/parse-tree-json-with-locs1.rb:3:1-3:9",
```

Comment:
Are you sure you don't want something like

```json
"loc": {
  "file": "test/cli/parse-tree-json-with-locs/parse-tree-json-with-locs1.rb",
  "start": {
    "line": 3,
    "col": 1,
  },
  "end": {
    "line": 3,
    "col": 9,
  },
},
```

Reply:
It's a good question. Since each node will now display the location I was going for a shorter version to make the whole output smaller. We can sacrifice size for ease of use and provide a pre-parser location.

Reply:
if you're interested in conserving size, the best thing do to would be to not re-construct a `core::Loc` at every node, and to instead store a top-level field like

```json
{
"file": "...",
"node": { ... }
}
```

there is a reason why we store only `core::LocOffset` on the nodes—it's 8 bytes instead of 16 bytes for a `core::Loc`.

Reply:
Let's keep `"loc" : "test/cli/parse-tree-json-with-locs/parse-tree-json-with-locs1.rb:3:1-3:9"` as proposed here. It's easy to parse, not too noisy so easy to read, and when reading long files or output in terminal it's easier to follow along.

The best thing would be to add customizations options to `print`:

```bash
sorbet --print symbol-table \
  --print-format "json" \  # json output
  --print-locs \ # show locs
  --print-locs-parsed \ # show line locs as { start: {...} end: {...} }
  --print-locs-unified \ # show file locs as { file: ... }
```